.github/workflows/client.yml

```
@@ -0,0 +1,30 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows 
+on: [push, pull_request]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true # ignore the failure of a step and avoid terminating the job
+      - name: Create test results directory
+        run: |
+          mkdir -p /tmp/test-results/keep-core-go
+      - name: Run Docker build
+        run: |
+          docker build --target gobuild --tag go-build-env .
+          docker build --tag keep-client . 
+      - name: Run Go tests
+        run: |
+          docker run --volume /tmp/test-results/keep-core-go:/mnt/test-results --workdir /go/src/github.com/keep-network/keep-core go-build-env gotestsum --junitfile /mnt/test-results/unit-tests.xml
+          docker run --volume /tmp/test-results/keep-core-go:/mnt/test-results --workdir /go/src/github.com/keep-network/keep-core go-build-env cat /mnt/test-results/unit-tests.xml > ./unit-tests.xml
```

Thoughts on breaking the lines a little bit? This could improve readability.

```suggestion
          docker run \
            --volume /tmp/test-results/keep-core-go:/mnt/test-results \
            --workdir /go/src/github.com/keep-network/keep-core \
            go-build-env \
            gotestsum --junitfile /mnt/test-results/unit-tests.xml
          docker run \
            --volume /tmp/test-results/keep-core-go:/mnt/test-results \
            --workdir /go/src/github.com/keep-network/keep-core \
            go-build-env \
            cat /mnt/test-results/unit-tests.xml > ./unit-tests.xml
```

---

Agreed, done.